pkg/terraform/install_test.go

```diff
@@ -26,34 +27,52 @@ func TestMixin_UnmarshalInstallStep(t *testing.T) {
 	b, err := ioutil.ReadFile("testdata/install-input.yaml")
 	require.NoError(t, err)
 
-	var step InstallStep
-	err = yaml.Unmarshal(b, &step)
+	var action InstallAction
+	err = yaml.Unmarshal(b, &action)
 	require.NoError(t, err)
+	require.Len(t, action.Steps, 1)
+	step := action.Steps[0]
 
 	assert.Equal(t, "Install MySQL", step.Description)
+	assert.Equal(t, true, step.Init)
```

Comment:
Is it safe to unconditionally init terraform every time? I don't know enough about it, so I'm just asking for my own curiosity.

Reply:
Good question.  I think it's becoming apparent that we _need_ to init every time, now that I'm reconciling terraform with how porter (or, generally, CNAB) bundles currently operate.  If I'm not mistaken, each porter/CNAB action is a fresh container run using the invocation image, with fresh context, and therefore in tf-land, we do need to init each time.

For maintaining/re-using state, I've created https://github.com/deislabs/porter-terraform/issues/4 for use of remote backends.  I'm not currently aware of a technique to re-use state in a CNAB context locally, yet.  I need to research this.

Which is to say, I think at this point, we probably need to assume that every action needs to `init` first (and, thus, the utility of allowing user to specify whether or not to via the porter manifest is not really useful/applicable.)

Reply:
Oh so init is a client side action?

Reply:
init downloads all terraform modules required to run

Reply:
Hmmm, what if we ran int as part of the docker build instead of at runtime? Is that an option?

Reply:
Oh, good idea!  This might definitely be the best way to go, once porter/mixins support it.  I'll file a ticket to revisit.